lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2014,8 +2014,20 @@ static bool isExpressionEmittedInline(Operation *op) {
   if (op->hasOneUse() && isa<hw::OutputOp>(*op->getUsers().begin()))
     return true;
 
+  // We don't want `hw.name` affecting inline vs not, so count the number of
+  // non-`hw.name` users and get the rest of this branch on that being more than
+  // 1. Make this a separate function since it is more expensive than the
+  // `hasOneUse()` check, so run it only if that check fails.
+  auto hasOneNonHwNameUse = [op]() {
+    size_t nonHwNameUserCount = 0;
+    for (auto *user : op->getResult(0).getUsers())
```

Comment:
this could early exit at the second use.  Then the short circuit below wouldn't be necessary.